src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -517,42 +598,71 @@ InvokerFactory MakeImplGemmDynamicForwardXdlopsNHWCInvokerFactory(
             const auto ker =
                 handle.Run(kernels[(isGfx90aFp16altSupport && data_ctx.gfx90aFp16alt) ? 1 : 0]);
             float elapsed = 0;
-            TensorDescriptor workspaceDesc(
-                miopenFloat, tensors.outDesc.GetLengths(), tensors.outDesc.GetStrides());
 
-            const bool need_cast = [&]() {
-                if(tensors.outDesc.GetType() == miopenHalf)
-                    return use_fp32_global_split_on_fp16;
-                if(tensors.outDesc.GetType() == miopenBFloat16)
-                    return need_set_zero;
-                return false;
-            }();
-
-            if(need_cast)
-            {
-                opArgs[0] = OpKernelArg(tensors.in);
-                opArgs[1] = OpKernelArg(tensors.w);
-                opArgs[2] = OpKernelArg(workSpace);
-            }
-            else
-            {
-                opArgs[0] = OpKernelArg(tensors.in);
-                opArgs[1] = OpKernelArg(tensors.w);
-                opArgs[2] = OpKernelArg(tensors.out);
-            }
+            auto trans_input_buf = trans_input_size == 0
+                                       ? null_buf
+                                       : const_cast<miopen::Handle*>(&handle)->CreateSubBuffer(
```

Comment:
Make `CreateSubBuffer()` a const method and get rid of `const_cast`.

[Note] The rule of thumb: dangerous stuff (like `const_cast`) should not be used unless absolutely necessary and without explanatory comments.

Reply:
OK, let me do this

Reply:
fixed